# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1322727343 18000
#      Thu Dec 01 03:15:43 2011 -0500
# Node ID 241f68fb48c336c63813b01aa399a53dffcd8423
# Parent  9ff0273cd0549dc5167b93cc341bc365f7541f8a
octave_map::cat: avoid indexing beyond end of array

* oct-map.cc (octave_map::cat (int, octave_idx_type, const
octave_map*)): Skip check for all fields being the same if there are
no fields.

diff --git a/src/oct-map.cc b/src/oct-map.cc
--- a/src/oct-map.cc
+++ b/src/oct-map.cc
@@ -750,24 +750,29 @@ octave_map::cat (int dim, octave_idx_typ
             {
               retval.xkeys = map_list[idx].xkeys;
               break;
             }
         }
 
       // Try the fast case.
       bool all_same = true;
-      for (octave_idx_type i = 0; i < n; i++)
+
+      if (nf > 0)
         {
-          all_same = map_list[idx].xkeys.is_same (map_list[i].xkeys);
-          if (! all_same)
-            break;
+          for (octave_idx_type i = 0; i < n; i++)
+            {
+              all_same = map_list[idx].xkeys.is_same (map_list[i].xkeys);
+
+              if (! all_same)
+                break;
+            }
         }
 
-      if (all_same)
+      if (all_same && nf > 0)
         do_cat (dim, n, map_list, retval);
       else
         {
           if (nf > 0)
             {
               // permute all structures to correct order.
               OCTAVE_LOCAL_BUFFER (octave_map, new_map_list, n);
 
